addons/l10n_es_edi_facturae/models/account_edi_format.py

```diff
@@ -0,0 +1,499 @@
+
+    def _l10n_es_edi_facturae_post_invoices(self, invoices):
+        # Ensure a certificate is available.
+        certificate = self.env['l10n_es_edi_facturae.certificate'].search([("company_id", 'in', invoices.company_id.ids)])
+        if not certificate:
+            return {inv: {
+                'error': _("Please configure the certificate for Facturae."),
+                'blocking_level': 'error',
+            } for inv in invoices}
+
+        # Generate the XMLs.
+        res = {}
+        for inv in invoices:
+            xml_name = f'{inv.name.replace("/", "_")}_facturae_signed.xml'
+            attachment = self.env['ir.attachment'].create({
+                'name': xml_name,
+                'raw': self.l10n_es_edi_facturae_render_facturae(inv),
+                'res_model': 'account.edi.document',
+                'mimetype': 'application/xml'
+            })
+            res[inv] = {'success': True, 'attachment': attachment}
+        return res
+
+    @api.model
+    def l10n_es_edi_facturae_clean_xml(self, tree_or_str, as_string=True):
+        """
+        Remove all the unwanted whitespaces and end of lines from the xml tree.
+        :param etree.XML or str tree_or_str: the xml tree in etree or str format
+        :return: the cleaned xml tree
+        """
+        str_tree = tree_or_str if isinstance(tree_or_str, str) else etree.tostring(tree_or_str, encoding=str)
+        str_tree = re.sub('\n+', '', str_tree)
+        str_tree = re.sub(r'\s{2, }', '', str_tree)
+        if as_string:
+            return str_tree
+        return etree.XML(str_tree)
+
+    @api.model
+    def l10n_es_edi_facturae_inv_lines_to_items(self, invoice, conv_rate, needs_equivalency, fmt_in_cur, fmt_in_eur):
+        """
+        Convert the invoice lines to a list of items required for the Facturae xml generation
+
+        :param account.move invoice:    The invoice containing the invoice lines
+        :param float conv_rate:         The conversion rate from the invoice currency to the company currency
+        :param bool needs_equivalency:  True if the invoice uses multi-currencies
+        :param func fmt_in_cur:         The formatting function to use for the amount in the invoice currency
+        :param func fmt_in_eur:         The formatting function to use for the amount in euros
+        :return tuple:                  A tuple containing the Face items, the taxes and the invoice totals data.
+        """
+        def compute_tax_amount(tax, line):
+            if tax.amount_type == 'percent':
+                return tax.amount * line.price_subtotal / 100
+            if tax.amount_type == 'fixed':
+                return tax.amount
+            if tax.amount_type == 'division':
+                if 1 + (tax.amount / 100):
+                    return line.price_subtotal - (line.price_subtotal / (1 + (tax.amount / 100)))
+                return 0.0
+            return 0.0
```

Comment:
Plz use the EDI helper to compute the taxes data... Are you living in a gingerbread world to think taxes computation is so easy?